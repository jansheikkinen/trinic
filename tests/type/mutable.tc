// Mutable

function main(void) -> void do
  let a: int8, b: mut int8, c: *int8, d: *mut int8,
      e: mut *int8, f: mut *mut int8 = undefined;
end
