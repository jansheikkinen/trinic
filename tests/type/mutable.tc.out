
### LEXER ###
000:     FUNCTION,     (null), (003, 008)
001:   IDENTIFIER,       main, (003, 013)
002:            (,     (null), (003, 014)
003:         VOID,     (null), (003, 018)
004:            ),     (null), (003, 019)
005:           ->,     (null), (003, 021)
006:         VOID,     (null), (003, 026)
007:           DO,     (null), (003, 029)
008:          LET,     (null), (004, 005)
009:   IDENTIFIER,          a, (004, 007)
010:            :,     (null), (004, 008)
011:         INT8,     (null), (004, 013)
012:            ,,     (null), (004, 014)
013:   IDENTIFIER,          b, (004, 016)
014:            :,     (null), (004, 017)
015:          MUT,     (null), (004, 021)
016:         INT8,     (null), (004, 026)
017:            ,,     (null), (004, 027)
018:   IDENTIFIER,          c, (004, 029)
019:            :,     (null), (004, 030)
020:            *,     (null), (004, 032)
021:         INT8,     (null), (004, 036)
022:            ,,     (null), (004, 037)
023:   IDENTIFIER,          d, (004, 039)
024:            :,     (null), (004, 040)
025:            *,     (null), (004, 042)
026:          MUT,     (null), (004, 045)
027:         INT8,     (null), (004, 050)
028:            ,,     (null), (004, 051)
029:   IDENTIFIER,          e, (005, 007)
030:            :,     (null), (005, 008)
031:          MUT,     (null), (005, 012)
032:            *,     (null), (005, 014)
033:         INT8,     (null), (005, 018)
034:            ,,     (null), (005, 019)
035:   IDENTIFIER,          f, (005, 021)
036:            :,     (null), (005, 022)
037:          MUT,     (null), (005, 026)
038:            *,     (null), (005, 028)
039:          MUT,     (null), (005, 031)
040:         INT8,     (null), (005, 036)
041:            =,     (null), (005, 038)
042:    UNDEFINED,     (null), (005, 048)
043:            ;,     (null), (005, 049)
044:          END,     (null), (006, 003)

### AST GENERATION ###
Declaration FUNCTION  main
  Stmt VARDECL
    Arg IDENTIFIER
      Identifier_Arg a
        Type TYPE INT8
      Identifier_Arg b
        Type TYPE INT8
          MUTABLE
      Identifier_Arg c
        Type POINTER
          Type TYPE INT8
      Identifier_Arg d
        Type POINTER
          Type TYPE INT8
            MUTABLE
      Identifier_Arg e
        Type POINTER
          Type TYPE INT8
          MUTABLE
      Identifier_Arg f
        Type POINTER
          Type TYPE INT8
            MUTABLE
          MUTABLE
    Arg EXPR
      Expr UNDEFINED
